tests/integration/new_relations/test_relations_coherence.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+import logging
+import secrets
+import string
+
+import psycopg2
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from ..helpers import CHARM_SERIES, DATABASE_APP_NAME
+from .helpers import build_connection_string
+from .test_new_relations import DATA_INTEGRATOR_APP_NAME
+
+logger = logging.getLogger(__name__)
+
+APPLICATION_APP_NAME = "postgresql-test-app"
+APP_NAMES = [DATABASE_APP_NAME, DATA_INTEGRATOR_APP_NAME]
+FIRST_DATABASE_RELATION_NAME = "first-database"
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_relations(ops_test: OpsTest, database_charm):
+    """Test that check relation data."""
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy(
+            database_charm,
+            application_name=DATABASE_APP_NAME,
+            num_units=1,
+            series=CHARM_SERIES,
+            config={"profile": "testing"},
+        )
```

Comment:
K8s charms also need an image resource you can use the [build_and_deploy](https://github.com/canonical/postgresql-k8s-operator/blob/main/tests/integration/helpers.py#L63-L70) helper to set that up.

Reply:
Oh, this is a reason why CI/CD failing, ACK: please update PR.